### DISCUSSION TITLE
Lightweight heartbeat test

### DISCUSSION THREAD

**Discussion by SnarfPunk** [alexjustesen/speedtest-tracker#565] [Ideas]
Would it be possible to add a secondary heartbeat test? I see frequent occasions where the speedtest executed on schedule failed, presumably because my internet service was actually down. I could set the frequency of the test much higher, but it would be fairly heavyweight to do so when just a simple ping would suffice. 

Essentially, I'd want to ping my ISP's gateway (and/or Google DNS, etc) every 60 seconds just to verify whether my connection to the outside world is alive, in addition to seeing the bandwidth stats of the speedtest on an hourly basis. 

Ideal functionality would include the following:

- Separate cron config for heartbeat test interval
- Configurable endpoint IP address or DNS name to ping
- Number of packets (default may be 1, but perhaps allow up to 5 and average?)
- Graph the data on dashboard

PS - I'm so happy I found this updated version - thank you for bringing the speed tracker back to life!

**Comment by alexjustesen** [comment]
All ready planning on it, although I'm calling is a "ping" but essentially the same feature. Scheduler for running tasks needs refactoring before I do it though so it's at least a few versions out.
